openhcl/page_pool_alloc/src/lib.rs

```diff
@@ -269,7 +582,66 @@ impl PagePool {
         }
     }
 
-    // TODO: save method and restore
+    /// Validate that all allocations have been restored. This should be called
+    /// after all devices have been restored.
+    ///
+    /// `leak_unrestored` controls what to do if a matching allocation was not
```

Comment:
Do we really have a use case when leaks are allowed? Like, saving memory for a device set but not all of them came back?

Reply:
That's a good question. I would imagine we _shouldn't_ allow leaks, but this is a policy toggle we can decide on. For now, because device drivers in this branch don't support save restore of allocations, I needed to allow leaks for it to work at all.

Reply:
We probably need to make sure the device drivers in this branch _free_ everything before servicing, then, right? We're not going to update all the drivers to support save/restore in all cases. And we don't want those to count as leaks.

Reply:
I think that's a good idea. That's more work, but I'll take a look at it tomorrow.

Reply:
OK if we get this in first and then fix up the drivers.

Reply:
#441